test/e2e/framework/volume/fixtures.go

```diff
@@ -696,3 +697,65 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 	return &v1.SELinuxOptions{
 		Level: "s0:c0,c1"}
 }
+
+// CheckVolumeModeOfPath check mode of volume
+func CheckVolumeModeOfPath(f *framework.Framework, pod *v1.Pod, volMode v1.PersistentVolumeMode, path string) {
+	if volMode == v1.PersistentVolumeBlock {
+		// Check if block exists
+		VerifyExecInPodSucceed(f, pod, fmt.Sprintf("test -b %s", path))
+
+		// Double check that it's not directory
+		VerifyExecInPodFail(f, pod, fmt.Sprintf("test -d %s", path), 1)
+	} else {
+		// Check if directory exists
+		VerifyExecInPodSucceed(f, pod, fmt.Sprintf("test -d %s", path))
+
+		// Double check that it's not block
+		VerifyExecInPodFail(f, pod, fmt.Sprintf("test -b %s", path), 1)
+	}
+}
+
+// PodExec runs f.ExecCommandInContainerWithFullOutput to execute a shell cmd in target pod
+func PodExec(f *framework.Framework, pod *v1.Pod, shExec string) (string, string, error) {
```

Comment:
Why are these pod exec functions a part of this volume package? From what I can tell, it does not contain any volume-specific logic. Is test/e2e/storage/utils/pod.go a better location? (I'm not sure, just spitballing an idea).

Reply:
There is a dependency issue here.
PodExec is used by "VerifyExecInPodSucceed" and "VerifyExecInPodFail" which are used by "CheckVolumeModeOfPath".

And this "CheckVolumeModeOfPath" is also used by e2evolume. So ideally we dont want to put them under storage because that will have a situation where parent package import subpackage.

I think ideally these function should rest in e2epod. But then there will be another dependency issue happen which is caused by : https://github.com/kubernetes/kubernetes/issues/81245

I think I will put this PodExec here for now and had another separate PR maybe to clear the pods dependency in framework so that we can move this over.